### PR TITLE
Feature/sc 3105/view to send funds from ethereum to optimism

### PR DESCRIPTION
**Pull request by Tulun**
View for the 3 different dropdown elements in the tickets below. Note: some of the data is stubbed as it's not yet ready (Boba).

https://app.shortcut.com/uma-project/story/3105/view-to-send-funds-from-ethereum-to-optimism

https://app.shortcut.com/uma-project/story/3107/view-to-send-funds-from-ethereum-to-boba

https://app.shortcut.com/uma-project/story/3106/view-to-send-funds-from-ethereum-to-arbitrum

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -378,13 +399,38 @@ export const CHAINS: Record<ChainId, ChainInfo> = {
...
+  [ChainId.MAINNET]: {
+    // Stubbed value. Does not work. TODO: Change this out when contract deployed.
+    BRIDGE: "0x2271a5E74eA8A29764ab10523575b41AA52455f0",
+  },
   [ChainId.RINKEBY]: {},
   [ChainId.KOVAN]: {},
-  [ChainId.OPTIMISM]: {},
+  [ChainId.OPTIMISM]: {
+    // Stubbed value. Does not work. TODO: Change this out when contract deployed.
+    BRIDGE: "0x2271a5E74eA8A29764ab10523575b41AA52455f0",
```
Comment (daywiss) [inline src/utils/constants.ts:428]: 0x3baD7AD0728f9917d1Bf08af5782dCbD516cDd96 based on the optimism deposit box [here](https://docs.google.com/document/d/1zAFxC3GDIy9Ig8Smu1Nr5Ubi81ROPNnWVuMOLOWjjKU/edit)

File: src/utils/constants.ts [hunk 2]
```diff
@@ -378,13 +399,38 @@ export const CHAINS: Record<ChainId, ChainInfo> = {
...
+      decimals: 18,
+    },
+  },
 };
 
 export const ADDRESSES: Record<ChainId, { BRIDGE?: string }> = {
-  [ChainId.MAINNET]: {},
+  [ChainId.MAINNET]: {
+    // Stubbed value. Does not work. TODO: Change this out when contract deployed.
+    BRIDGE: "0x2271a5E74eA8A29764ab10523575b41AA52455f0",
```
Comment (daywiss) [inline src/utils/constants.ts:422]: i wonder if this should not have a value? theres no deposit box on mainnet @mrice32 ?

File: src/utils/constants.ts [hunk 3]
```diff
@@ -440,6 +486,13 @@ export const PROVIDERS: Record<ChainId, GetProvider> = {
         `https://arbitrum-rinkeby.infura.io/v3/${process.env.REACT_APP_PUBLIC_INFURA_ID}`
       )
   ),
+  // Doesn't have an rpc on infura.
+  [ChainId.BOBA]: memoize(
+    () =>
+      new ethers.providers.StaticJsonRpcProvider(
+        `https://arbitrum-rinkeby.infura.io/v3/${process.env.REACT_APP_PUBLIC_INFURA_ID}`
```
Comment (daywiss) [inline src/utils/constants.ts:493]: would this work? https://mainnet.boba.network/ see https://docs.boba.network/developer-docs/006_parameters-mainnet and https://docs.boba.network/developer-docs/005_parameters-rinkeby

File: src/utils/constants.ts [hunk 4]
```diff
@@ -530,3 +583,65 @@ export const GAS_PRICE_BUFFER = toWeiSafe(
 );
 // Rounded up from a mainnet transaction sending eth gas limit
 export const ADD_LIQUIDITY_ETH_GAS = ethers.BigNumber.from(82796);
+
+export const CHAINS_SELECTION = [
+  {
+    name: "Optimistic",
```
Comment (daywiss) [inline src/utils/constants.ts:589]: i think should be Optimism

File: src/utils/constants.ts [hunk 5]
```diff
@@ -530,3 +583,65 @@ export const GAS_PRICE_BUFFER = toWeiSafe(
...
+    chainId: ChainId.MAINNET,
+    logoURI: ethereumLogo,
+    // Doesn't have an RPC on Infura. Need to know how to handle this
+    rpcUrl: "https://mainnet.infura.io/v3/",
+    explorerUrl: "https://etherscan.io",
+    constructExplorerLink: (txHash: string) =>
+      `https://etherscan.io/tx/${txHash}`,
+    nativeCurrency: {
+      name: "Boba",
+      symbol: "BOBA",
```
Comment (daywiss) [inline src/utils/constants.ts:643]: typo here

File: src/utils/constants.ts [hunk 6]
```diff
@@ -530,3 +583,65 @@ export const GAS_PRICE_BUFFER = toWeiSafe(
...
+
+export const CHAINS_SELECTION = [
+  {
+    name: "Optimistic",
+    chainId: ChainId.OPTIMISM,
+    logoURI: optimismLogo,
+    rpcUrl: "https://mainnet.optimism.io",
+    explorerUrl: "https://optimistic.etherscan.io",
+    constructExplorerLink: defaultConstructExplorerLink(
+      "https://optimistic.etherscan.io"
```
Comment (daywiss) [inline src/utils/constants.ts:595]: this function looks different from the others

**Review by daywiss — COMMENTED** [review]
works great, i think lets try to get as many valid addresses and constants as we can

File: src/utils/constants.ts (hunk 3 above)
Comment (Tulun) [reply src/utils/constants.ts:491]: Looks like it might work?

**Comment by daywiss** [comment]
when selecting boba in `from` dropdown, before you have a chance to switch network, it looks like balances are fetched: 
image
The other tokens dont seem to have this issue

**Comment by daywiss** [comment]
send fails with optimism 
image

**Comment by daywiss** [comment]
random crash when switching form optimism to arbitrum. not sure how to reproduce yet
image

**Comment by daywiss** [comment]
crashes when doing this:
1. select optimism and connect to network
2. change your coin selection to weth
3. switch to arbitrum. might crash. if not switch back to optimism

image

**Comment by Tulun** [comment]
> crashes when doing this:
> 
>     1. select optimism and connect to network
> 
>     2. change your coin selection to weth
> 
>     3. switch to arbitrum. might crash. if not switch back to optimism
> 
> 
> image



> crashes when doing this:
> 
>     1. select optimism and connect to network
> 
>     2. change your coin selection to weth
> 
>     3. switch to arbitrum. might crash. if not switch back to optimism
> 
> 
> image

This is a bug unrelated to the PR specifically: as it's a big enough refactor of the redux state, I would suggest we handle it in its own PR.

**Comment by daywiss** [comment]
heres an issue:
1. be disconected from site
2. connect to aribitrum in metamask
3. connect to site, see optimism selected, but see no banner asking you to change networks

**Comment by daywiss** [comment]
this crashes the app:
1. connect to site, select mainnet "from" and join network
2. switch currency to usdc
3. switch dropdown to "mainnet", see site crash

File: src/components/AddressSelection/AddressSelection.tsx [hunk 7]
```diff
@@ -1,31 +1,79 @@
...
+  const sendState = useAppSelector((state) => state.send);
+
+  // When redux state changes, make sure local inputs change.
+  // useEffect(() => {
+  //   if (sendState.fromChain === ChainId.MAINNET) {
+  //     setCurrentlySelectedChain(CHAINS_SELECTION[0]);
+  //   } else {
+  //     setCurrentlySelectedChain(CHAINS_SELECTION[2]);
+  //   }
+  // }, [sendState.fromChain]);
```
Comment (daywiss) [inline src/components/AddressSelection/AddressSelection.tsx:51]: should this still be in here?

File: src/utils/constants.ts [hunk 8]
```diff
@@ -261,6 +263,41 @@ export const TOKENS_LIST: Record<ChainId, TokenList> = {
       bridgePool: "",
     },
   ],
+  // Stubbed
+  [ChainId.BOBA]: [
+    {
+      address: getAddress("0xB47e6A5f8b33b3F17603C83a0535A9dcD7E32681"),
```
Comment (daywiss) [inline src/utils/constants.ts:269]: make sure we in the card to enable boba we go through and fix these addresses, some are right some are wrong. add this link to card https://blockexplorer.boba.network/tokens

File: src/components/AddressSelection/AddressSelection.tsx (hunk 7 above)
Comment (Tulun) [reply src/components/AddressSelection/AddressSelection.tsx]: Removed.

File: src/utils/constants.ts (hunk 8 above)
Comment (Tulun) [reply src/utils/constants.ts:269]: > this crashes the app:
> 
>     1. connect to site, select mainnet "from" and join network
> 
>     2. switch currency to usdc
> 
>     3. switch dropdown to "mainnet", see site crash

Same bug as the WETH.

**Review by daywiss — APPROVED** [review]
looks good. we are going to push a few issues to a seperate pr
1. add boba and boba send
2. add optimism send and explorer construction
3. fix crashing when selecting tokens and switching chains

**Pull request merged**